### ISSUE TITLE
q3map2 dynamic lighting can use more than one lightmap stage in a material

### ISSUE THREAD

**Issue by illwieckz** [DaemonEngine/Daemon#221]
Recently in `master` some code written by me was merged, this code raises a warning when more than one lightmap stage is found in a material definition. This was based on the wrong assumption than more than one lightmap stage in a material would be a mistake.

I found one legacy use case where it's not a mistake: q3map2-generated `q3map2_<basename>.shader` files describing dynamic lightmaps may have more than one lightmap stage in their material definitions. The Metro map by KOsAD makes use of this feature, as an example.

**Comment by illwieckz** [comment]
Example:

```c
metro/2f496655c5c341baef028e5e3b5513b7
{ // Q3Map2 defaulted
        {
                map $lightmap
                rgbGen identity
        }
        
        // Q3Map2 custom lightstyle stage(s)
        {
                map $lightmap
                blendFunc GL_SRC_ALPHA GL_ONE
                rgbGen wave triangle 0 .5 0 .8 // style 2
                tcGen lightmap
                tcMod transform 1 0 0 1 0.00000 0.50000
        }
        {
                map $lightmap
                blendFunc GL_SRC_ALPHA GL_ONE
                rgbGen wave triangle 0 .6 .3 .8 // style 3
                tcGen lightmap
                tcMod transform 1 0 0 1 0.00000 0.11719
        }
        {
                map $lightmap
                blendFunc GL_SRC_ALPHA GL_ONE
                rgbGen wave square 0 .5 .4 .6 // style 4
                tcGen lightmap
                tcMod transform 1 0 0 1 0.00000 0.23438
        }

        {
                map textures/metro/rfsp-stex5b
                blendFunc GL_DST_COLOR GL_ZERO
                rgbGen identity
        }
}
```

**Comment by bmorel** [comment]
So? What is current behavior, and what is expected behavior?

**Comment by illwieckz** [comment]
Fixed in da9bd8c0a8c13659b604e968899497cf16a38a55.
